dogpile/cache/backends/memcached.py

```diff
@@ -509,15 +555,32 @@ def __init__(self, arguments):
         self.serde = arguments.get("serde", pymemcache.serde.pickle_serde)
         self.default_noreply = arguments.get("default_noreply", False)
         self.tls_context = arguments.get("tls_context", None)
+        self.socket_keepalive = arguments.get("socket_keepalive", None)
+        self.retry_client = arguments.get("retry_client", False)
+        self.retry_attempts = arguments.get("retry_attempts", 2)
+        self.retry_delay = arguments.get("retry_delay", 0)
+        self.retry_for = arguments.get("retry_for", None)
+        self.do_not_retry_for = arguments.get("do_not_retry_for", None)
 
     def _imports(self):
         global pymemcache
         import pymemcache
 
     def _create_client(self):
-        return pymemcache.client.hash.HashClient(
+        client = pymemcache.client.hash.HashClient(
             self.url,
             serde=self.serde,
             default_noreply=self.default_noreply,
             tls_context=self.tls_context,
+            socket_keepalive=self.socket_keepalive,
         )
+        if self.retry_client:
+            return pymemcache.client.retry.RetryingClient(
+                client,
+                attempts=self.retry_attempts,
+                retry_delay=self.retry_delay,
+                retry_for=self.retry_for,
+                do_not_retry_for=self.do_not_retry_for
+            )
```

Comment:
any interest in tests?  We can add a new class to test_memcached_backend that includes these configs:


```py
class PyMemcacheRetryTest(_NonDistributedMemcachedTest):
    backend = "dogpile.cache.pymemcache"
    config_args = {"arguments": {"url": MEMCACHED_URL, "enable_retry_client": True, "retry_attempts": 3}}
```

just to make sure it round trips normally at least

Reply:
Yes, I planned to discuss this point with you, I wasn't sure how to implement this scenario within dogpile's testing context.

Thanks for these details I'll submit a new patch set.

Reply:
Done